var/spack/repos/builtin/packages/papi/package.py

```diff
@@ -47,7 +47,41 @@ class Papi(Package):
     version('5.4.1', '9134a99219c79767a11463a76b0b01a2')
     version('5.3.0', '367961dd0ab426e5ae367c2713924ffb')
 
+    variant('components',
+        default='',
+        values=('', 'example', 'cuda', 'nvml', 'infiniband',
+            'infiniband_umad', 'powercap', 'rapl', 'lmsensors'),
+        multi=True,
+        description='Include optional components')
+    depends_on('cuda', when='components=cuda')
+    depends_on('cuda', when='components=nvml')
+    depends_on('lm-sensors', when='components=lmsensors')
+
     def install(self, spec, prefix):
+        with working_dir("src/components/cuda"):
+            if 'components=cuda' in spec:
```

Comment:
I would put the if-statements outside of the with-statements. No need to change directories if the particular component isn't requested.

Reply:
Thanks for the recommendation.  I've modified this as suggested.